src/Events/Authorization/SubscriptionXacmlMapper.cs

```diff
@@ -0,0 +1,97 @@
+using System.Collections.Generic;
+using System.Security.Claims;
+
+using Altinn.Authorization.ABAC.Xacml.JsonProfile;
+using Altinn.Common.PEP.Constants;
+using Altinn.Common.PEP.Helpers;
+using Altinn.Platform.Events.Models;
+
+using static Altinn.Authorization.ABAC.Constants.XacmlConstants;
+
+namespace Altinn.Platform.Events.Authorization
+{
+    /// <summary>
+    /// Utility class for converting Events Subscription requests to XACML request
+    /// </summary>
+    public static class SubscriptionXacmlMapper
+    {
+        private const string DefaultIssuer = "Altinn";
+        private const string DefaultType = "string";
+
+        private const string PartyPrefix = "/party/";
+
+        private const string ClaimPartyID = "urn:altinn:partyid";
+
+        /// <summary>
+        /// Create a decision request based on a subscription
+        /// </summary>
+        public static XacmlJsonRequestRoot CreateDecisionRequest(Subscription subscription)
+        {
+            XacmlJsonRequest request = new()
+            {
+                AccessSubject = new List<XacmlJsonCategory>(),
+                Action = new List<XacmlJsonCategory>(),
+                Resource = new List<XacmlJsonCategory>()
+            };
+
+            string org = null;
+            string app = null;
+
+            string[] pathParams = subscription.SourceFilter.AbsolutePath.Split("/");
+
+            if (pathParams.Length > 2)
```

Comment:
we don't have an else case we can handle. If the source is not in the expected format. An absolute url to an app with the org/app to follow we don't know how to pick the parameters out, and the PDP will have to work withthe information is has available through the remaining parameters in the subscription. 

Same set up as for xacml mappers in other platform components I believe

Reply:
I guess my point is that I suspect that the check will never fail and that it is unnecessary.

In the off change that it can fail, what kind of authorization will it produce with app and org being empty? Is it plausible for the authorization to succeed?

Reply:
Lets leave this as is for now. I see there is a similar check in CloudEventXacmlMapper.CreateDecisionRequest()